sdks/python/setup.py

```diff
@@ -411,11 +411,13 @@ def get_portability_package_data():
               # Pinning docutils as a workaround for Sphinx issue:
               # https://github.com/sphinx-doc/sphinx/issues/9727
               'docutils==0.17.1',
+              'jinja2>=3.0,<3.1',
               'pandas<2.1.0',
           ],
           'test': [
               'docstring-parser>=0.15,<1.0',
               'freezegun>=0.3.12',
+              'jinja2>=3.0,<3.1',
```

Comment:
What's the reason for the upper bound cap given that newer versions are available: https://pypi.org/project/Jinja2/#history

See also:

https://cwiki.apache.org/confluence/display/BEAM/Dependency+management+guidelines+for+Beam+Python+SDK+maintainers